### DISCUSSION TITLE
How to exclude/remove check?

### DISCUSSION THREAD

**Discussion by ollimeier** [fonttools/fontbakery#3577] [Q&A]
Hi all,

may someone hep me with my dummy question? How do I exclude/remove a test?

I would love to extend Fontwerk's profile and make use of most of the googlefonts checks, but not all of some. I would like to remove a view.

This is what I currently try to do:
https://github.com/moontypespace/fontbakery/blob/0c9bfe877917654c85e786b09c9da26080045be2/Lib/fontbakery/profiles/fontwerk.py#L74

But it does not work for me.

any help is much appreciated.

Thanks a lot in advance, 
Olli

**Comment by ollimeier** [comment]
@felipesanches or @graphicore I am sure you guys have a nice solution for that, but I simply overlooked it.

**Comment by graphicore** [comment]
Since you are calling `profile.auto_register` I would use its `filter_func` argument:

https://github.com/googlefonts/fontbakery/blob/18ffa37c00eee8fd2107e50a0e4263036ed9dfd6/Lib/fontbakery/profile.py#L845-L874

Something like this (did not test):

```py
profile.auto_register(globals(), filter_func=lambda type, id: not (type == 'check' and id in CHECKS_DONT_DO));
```

However, I'm not sure why `profile.remove_check` doesn't work, it might be a bug.

**Comment by graphicore** [comment]
I just tried out both versions, both work so far, but the `filter_func` version needs to accept a third argument:

```py
profile.auto_register(globals(), filter_func=lambda type, id, _: not (type == 'check' and id in CHECKS_DONT_DO))
```

otherwise you get an Error:

```
TypeError: <lambda>() takes 2 positional arguments but 3 were given
```

`profile.remove_check` also works as intended, with both versions I get the Error:

```
File "/home/commander/Projects/googlefonts/fontbakery/Lib/fontbakery/profiles/fontwerk.py", line 108, in <module>
    profile.test_expected_checks(FONTWERK_PROFILE_CHECKS, exclusive=True)
  File "/home/commander/Projects/googlefonts/fontbakery/Lib/fontbakery/profile.py", line 325, in test_expected_checks
    raise SetupError(
fontbakery.errors.SetupError: Profile fails expected checks test:
missing checks: com.google.fonts/check/family/italics_have_roman_counterparts, com.google.fonts/check/canonical_filename, com.google.fonts/check/vendor_id;
```

Which is the expected behavior when in

```py
profile.test_expected_checks(FONTWERK_PROFILE_CHECKS, exclusive=True)
```

the `FONTWERK_PROFILE_CHECKS` list still contains the removed/filtered checks. To fail about this is the only purpose of `profile.test_expected_checks`.

I would thus suggest to construct the list like so:

```py
FONTWERK_PROFILE_CHECKS = \
    [check for check in GOOGLEFONTS_PROFILE_CHECKS
                if check not in CHECKS_DONT_DO] + [
        'com.fontwerk/check/no_mac_entries',
        'com.fontwerk/check/vendor_id',
        'com.fontwerk/check/weight_class_fvar',
    ]
```

or not to run `profile.test_expected_checks` in which case `FONTWERK_PROFILE_CHECKS` is not required either.

**Reply by ollimeier** [reply]
@graphicore  Thanks a lot Lasse for having a look into it. I did it the way you suggessed, without success:
https://github.com/moontypespace/fontbakery/commit/594e79c09982bd978d0f2e1d610b940f4472676a

I get the following:
```
  File "/Users/ollimeier/Documents/GitHub/olli/fontbakery/Lib/fontbakery/profiles/fontwerk.py", line 106, in <module>
    profile.test_expected_checks(FONTWERK_PROFILE_CHECKS, exclusive=True)
  File "/Users/ollimeier/Documents/GitHub/olli/fontbakery/Lib/fontbakery/profile.py", line 325, in test_expected_checks
    raise SetupError(
fontbakery.errors.SetupError: Profile fails expected checks test:
unexpected checks: com.google.fonts/check/family/italics_have_roman_counterparts, com.google.fonts/check/fstype, com.google.fonts/check/gasp, com.google.fonts/check/canonical_filename, com.google.fonts/check/vendor_id;
```

**Comment by graphicore** [comment]
@moontypespace In moontypespace@594e79c you don't apply the `filter_func` or `remove_check`, hence the checks are not removed from the profile.
